packages/flutter/test/material/bottom_navigation_bar_test.dart

```diff
@@ -1976,7 +2329,63 @@ void main() {
         }
         await expectLater(
           find.byType(BottomNavigationBar),
-          matchesGoldenFile('bottom_navigation_bar.shifting_transition.${pump - 1}.png'),
+          matchesGoldenFile('m2_bottom_navigation_bar.shifting_transition.${pump - 1}.png'),
+        );
+      });
+    }
+  });
+
+  group('Material3 - BottomNavigationBar shifting backgroundColor with transition', () {
+    // Regression test for: https://github.com/flutter/flutter/issues/22226
+    Widget runTest() {
+      int currentIndex = 0;
+      const bool useInkSparkle = !kIsWeb;
```

Comment:
Why not on web?

Reply:
I am pretty sure web supports shaders now, I wonder if we need to update the InkSparkle then...

Reply:
I don't think `InkSparkle` shader would work on web + Android mobile (web might support shaders but it is probably on CanvasKit). But we might need to ask someone who knows more.

`InkSparkle` documentation explicitly mentionned it is not supported with the HTML renderer, see
https://github.com/flutter/flutter/blob/6e9d4ccc250dd56ad967de221258d2b8eeffe8e3/packages/flutter/lib/src/material/ink_sparkle.dart#L17-L18

As of today `InkSparkle` is deactived on Web. For instance, the `useInkSparkle` variable I used in this test is direcly inspired by the `ThemeData` implementation, see https://github.com/flutter/flutter/blob/6e9d4ccc250dd56ad967de221258d2b8eeffe8e3/packages/flutter/lib/src/material/theme_data.dart#L418

And tests for `InkSparkle` are deactivated on Web, see https://github.com/flutter/flutter/blob/6e9d4ccc250dd56ad967de221258d2b8eeffe8e3/packages/flutter/test/material/ink_sparkle_test.dart#L33

Reply:
(We could also file a bug about this and follow up in a separate change 🙂 )

Reply:
FYI, I filed https://github.com/flutter/flutter/issues/138487.